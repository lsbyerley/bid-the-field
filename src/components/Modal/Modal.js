import { Fragment, useRef } from 'react';
import { Dialog, Transition } from '@headlessui/react';
import { useAppContext } from '../../AppContext';
import ReactMarkdown from 'react-markdown';
import remarkGfm from 'remark-gfm';

const Modal = () => {
  let closeButtonRef = useRef(null);
  const { modalOpen, setModalOpen, modalContent, setModalContent } =
    useAppContext();

  const closeModal = () => {
    setModalOpen(false);
    // timeout to wait for modal transition to finish
    setTimeout(() => setModalContent(''), 500);
  };

  return (
    <Transition appear show={modalOpen} as={Fragment}>
      <Dialog
        as='div'
        className='fixed inset-0 z-10 overflow-y-auto'
        onClose={closeModal}
        initialFocus={closeButtonRef}
      >
        <div className='modal modal-open'>
          <Transition.Child
            as={Fragment}
            enter='ease-out duration-300'
            enterFrom='opacity-0'
            enterTo='opacity-100'
            leave='ease-in duration-200'
            leaveFrom='opacity-100'
            leaveTo='opacity-0'
          >
            <Dialog.Overlay className='fixed inset-0' />
          </Transition.Child>

          <Transition.Child
            as={Fragment}
            enter='ease-out duration-300'
            enterFrom='opacity-0 scale-95'
            enterTo='opacity-100 scale-100'
            leave='ease-in duration-200'
            leaveFrom='opacity-100 scale-100'
            leaveTo='opacity-0 scale-95'
          >
            <div className='modal-box'>
              <button
                className='absolute btn btn-sm btn-circle right-2 top-2'
                onClick={() => closeModal()}
                ref={closeButtonRef}
              >
                âœ•
              </button>
              <div className='mt-8 prose'>
                <ReactMarkdown
                  children={modalContent}
                  remarkPlugins={[remarkGfm]}
                />
              </div>
            </div>
          </Transition.Child>
        </div>
      </Dialog>
    </Transition>
  );
};

export default Modal;
