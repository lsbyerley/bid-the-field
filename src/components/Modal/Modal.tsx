import { Fragment, useRef } from 'react';
import { Dialog, Transition, TransitionChild } from '@headlessui/react';
import { useAppContext } from '@/AppContext';
import ReactMarkdown from 'react-markdown';
import remarkGfm from 'remark-gfm';

const Modal = () => {
  const { modalOpen, setModalOpen, modalContent, setModalContent } =
    useAppContext();

  const closeModal = () => {
    setModalOpen(false);
    // timeout to wait for modal transition to finish
    setTimeout(() => setModalContent(''), 500);
  };

  return (
    <Transition appear show={modalOpen} as={Fragment}>
      <Dialog
        as='div'
        className='fixed inset-0 z-10 overflow-y-auto'
        onClose={() => closeModal()}
        open={modalOpen}
      >
        <div className='modal modal-open'>
          <TransitionChild
            as={Fragment}
            enter='ease-out duration-300'
            enterFrom='opacity-0'
            enterTo='opacity-100'
            leave='ease-in duration-200'
            leaveFrom='opacity-100'
            leaveTo='opacity-0'
          >
            <div className='fixed inset-0 bg-black/30'  aria-hidden='true' />
          </TransitionChild>

          <TransitionChild
            as={Fragment}
            enter='ease-out duration-300'
            enterFrom='opacity-0 scale-95'
            enterTo='opacity-100 scale-100'
            leave='ease-in duration-200'
            leaveFrom='opacity-100 scale-100'
            leaveTo='opacity-0 scale-95'
          >
            <div className='modal-box'>
              <button
                className='absolute btn btn-sm btn-circle right-2 top-2'
                onClick={() => closeModal()}
              >
                âœ•
              </button>
              <div className='mt-8 prose'>
                <ReactMarkdown
                  children={modalContent}
                  remarkPlugins={[remarkGfm]}
                />
              </div>
            </div>
          </TransitionChild>
        </div>
      </Dialog>
    </Transition>
  );
};

export default Modal;
