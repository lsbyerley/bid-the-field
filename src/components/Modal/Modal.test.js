import React from 'react';
import { render, screen } from 'test-utils';
import Modal from './Modal';

jest.mock('@headlessui/react', () => ({
  Dialog: jest.fn(({ props }) => <div>Dialog {JSON.stringify(props)}</div>),
  'Dialog.Overlay': jest.fn(({ props }) => (
    <div>DialogOverlay {JSON.stringify(props)}</div>
  )),
  Transition: jest.fn(({ props }) => (
    <div>Transition {JSON.stringify(props)}</div>
  )),
  'Transition.Child': jest.fn(({ props }) => (
    <div>TransitionOverlay {JSON.stringify(props)}</div>
  )),
}));

jest.mock('react-markdown', () => ({
  __esModule: true,
  default: (props) => {
    return <div data-testid='react-markdown'>{JSON.stringify(props)}</div>;
  },
}));

jest.mock('remark-gfm', () => ({
  default: jest.fn(),
}));

jest.mock('@/AppContext', () => ({
  useAppContext: () => ({
    modalOpen: true,
    setModalOpen: jest.fn(),
    modalContent: 'test',
    setModalContent: jest.fn(),
  }),
}));

const renderModal = (props) => {
  const utils = render(<Modal {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('Modal', () => {
  it('should render and match snapshot', () => {
    const { home } = renderModal();
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
