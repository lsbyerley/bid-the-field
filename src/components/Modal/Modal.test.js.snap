// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render and match snapshot 1`] = `
<div>
  Transition 
</div>
`;
